### DISCUSSION TITLE
Toolbars for non-markdown files

### DISCUSSION THREAD

**Discussion by azaol-aegnor** [chrisgurney/obsidian-note-toolbar#130] [Ideas]
Hello!

In a folder that may contain pictures, PDF files or other types of file, we would love to be able to display a Note Toolbar.

I'll give a couple of real world use cases. Both use a plugin that add a command to go to the next/previous file listed in the explorer:
1. I have a folder with everything related to a trip, clicking next/previous would allow to swap from my boarding PDF to my partner's one.
2. I have a folder with a few pictures and I'd like to browse the quickly on mobile without having to pull the explorer and select the next one individually. Considering I also miss click sometimes. Clicking next/previous is much more convinient!

In bothcases, because the file is not a Markdown file, the toolbar does not show up. Could we have a checkbox/option on a mapping to specify the toolbar can be display on any type of file (or a specific given set)?

**Comment by chrisgurney** [comment]
Thank you for the idea! Yes I'm open to adding this, as I discovered that it _also_ enables other use cases (particularly canvases).

In the meantime, have you looked at the [Quick Explorer](https://github.com/pjeby/quick-explorer) plugin? It includes commands for previous/next file navigation that you can bind to hotkeys (though I recognize that this is impractical on a phone).

Once I add support for toolbars on these scenarios, that plugin will probably be my recommendation to add the necessary commands.

**Reply by azaol-aegnor** [reply]
Hey!

Thanks for answering this fast! Looking forward the update 😇


I did see the plugin and as it's not available on mobile, as pointed out, I coded a 30lines long one just for those two actions for my mom, along with a dozen of unpublished plugins I manage for family and friends. 😅

It would make sense to recommend that one. It would be ideal if their maintainer could test it on mobile and toggle the property in the manifest as well as *perhaps* improve the handling of the Folder Note concept (the current implementation in Quick Explorer assumes the note is named as the folder, but I believe the Note Folder plugins allows other configuration options). 
However, that is a while other topic!

Let me know if you need tester for the toolbar or if I can help somehow! Thanks again!

**Comment by chrisgurney** [comment]
See also #78 and #200

**Comment by chrisgurney** [comment]
@azaol-aegnor Support for toolbars appearing in non-markdown files (audio, images, PDFs, and video) is now available [in the latest beta](https://github.com/chrisgurney/obsidian-note-toolbar/discussions/246). 

If you give it a try, let me know how it works for you!

**Reply by azaol-aegnor** [reply]
I will once I'm done moving in a few days!
Took a couple of minute to look at the beta announcement and it mentions PDF, audio and video, but modified my local copy of the plugin to display the bar on JSON and YAML files.
I'll still download it to take a look if the beta is not over by the time I get desk and boot my computer!

**Reply by chrisgurney** [reply]
@azaol-aegnor Thanks! If it's easier for you, just wait a few days and this should be out of beta.

**Reply by chrisgurney** [reply]
@azaol-aegnor I just re-read your note. Those are interesting file types. What do you have in toolbars for JSON and YAML files? And how did you modify Obsidian so you can view/edit those file types?

**Reply by azaol-aegnor** [reply]
Ho my, I don't know how I missed that notification, sorry,
I'm using the plugin called "VSCode Editor". Or at least I was! With Base, I got into centralizing more things into md files. It had me changed stuff on my personnal vault a little.

Short answer is: the plugin allows to specify file extension to be opened within Obsidian. As for the tool bar, the two most used buttons on mobile are to navigate to previous and next file in a given folder. A trip or photo album were two good examples, and game notes+canvas, or client meeting notes and assets are two others.

Down the road, I plan to also just share my phone screen on a TV when visiting family. So a couple of floating buttons to switch from a document to another are very handy.

Then, I have some custom scripts on some buttons to extract a set/part of the frontmatter properties into any suitable format (individual component properties to a well written (mail format) address, jump to a related view or file, call an API I run locally (on the phone or homelab), ...
